src/interchange/src/json.rs

```diff
@@ -8,10 +8,203 @@
+                Some(column_names_and_types(desc))
+            } else {
+                None
+            },
+            value_columns: column_names_and_types(value_desc),
+        }
+    }
+
+    pub fn encode_row(&self, row: repr::Row, names_types: &[(ColumnName, ColumnType)]) -> Vec<u8> {
+        let value = encode_datums_as_json(row.iter(), names_types);
+        value.to_string().into_bytes()
+    }
+}
+
+impl Encode for JsonEncoder {
+    fn get_format_name(&self) -> &str {
+        "json"
+    }
+
+    fn encode_key_unchecked(&self, row: repr::Row) -> Vec<u8> {
+        self.encode_row(
+            row,
+            self.key_columns.as_ref().expect("key schema must exist"),
+        )
+    }
+
+    fn encode_value_unchecked(&self, row: repr::Row) -> Vec<u8> {
+        self.encode_row(row, &self.value_columns)
+    }
+}
+
+impl fmt::Debug for JsonEncoder {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        f.debug_struct("JsonEncoder")
+            .field(
+                "schema",
+                &format!("{:?}", build_row_schema_json(&self.value_columns, "schema")),
+            )
+            .finish()
+    }
+}
+
+/// Encodes a sequence of `Datum` as JSON, using supplied column names and types.
+pub fn encode_datums_as_json<'a, I>(
+    datums: I,
+    names_types: &[(ColumnName, ColumnType)],
+) -> serde_json::value::Value
+where
+    I: IntoIterator<Item = Datum<'a>>,
+{
+    let mut name_idx = 0;
+    let value_fields = names_types
+        .iter()
+        .zip_eq(datums)
+        .map(|((name, typ), datum)| {
+            let name = name.as_str().to_owned();
+            (
+                name,
+                TypedDatum::new(datum, typ.clone()).json(&mut move || {
+                    let ret = format!("record{}", name_idx);
+                    name_idx += 1;
+                    ret
+                }),
+            )
+        })
+        .collect();
+    serde_json::value::Value::Object(value_fields)
+}
+
+pub trait ToJson {
+    /// Transforms this value to a JSON value.
+    fn json<F: FnMut() -> String>(self, namer: &mut F) -> serde_json::value::Value;
+}
+
+impl<'a> ToJson for TypedDatum<'_> {
+    fn json<F: FnMut() -> String>(self, namer: &mut F) -> serde_json::value::Value {
+        let TypedDatum { datum, typ } = self;
+        if typ.nullable && datum.is_null() {
+            serde_json::value::Value::Null
+        } else {
+            match &typ.scalar_type {
+                ScalarType::Bool => json!(datum.unwrap_bool()),
+                ScalarType::Int16 => json!(datum.unwrap_int16()),
+                ScalarType::Int32 | ScalarType::Oid => json!(datum.unwrap_int32()),
+                ScalarType::Int64 => json!(datum.unwrap_int64()),
+                ScalarType::Float32 => json!(datum.unwrap_float32()),
+                ScalarType::Float64 => json!(datum.unwrap_int64()),
+                ScalarType::Numeric { .. } => json!(datum.unwrap_numeric().0),
```

Comment:
Unfortunately this produces this bit:

```
"column5":{"bits":0,"digits":8,"exponent":-4,"lsu":[678,345,12,0,0,0,0,0,0,0,0,0,0]}
```

Which appears to be some internal representation -- it is neither human-readable nor conforms to the AVRO spec.